fmriprep/cli/run.py

```diff
@@ -245,14 +246,53 @@ def main():
     """Entry point"""
     from nipype import logging as nlogging
     from multiprocessing import set_start_method, Process, Manager
-    from .. import __version__
     from ..viz.reports import generate_reports
     from ..utils.bids import write_derivative_description
     set_start_method('forkserver')
 
     warnings.showwarning = _warn_redirect
     opts = get_parser().parse_args()
 
+    sentry_sdk = None
+    if not opts.notrack:
+        import sentry_sdk
+        from ..__about__ import __version__
+        environment = "prod"
+        if bool(int(os.getenv('FMRIPREP_DEV', 0))) or ('+' in __version__):
+            environment = "dev"
+
+        def before_send(event, hints):
+            # Filtering log messages about crashed nodes
+            if 'logentry' in event and 'message' in event['logentry']:
+                msg = event['logentry']['message']
+                if msg.startswith("could not run node:"):
+                    return None
+                elif msg.startswith("Saving crash info to "):
+                    return None
+                elif re.match("Node .+ failed to run on host .+", msg):
+                    return None
+            else:
+                return event
+
+        sentry_sdk.init("https://d5a16b0c38d84d1584dfc93b9fb1ade6@sentry.io/1137693",
+                        release=__version__,
+                        environment=environment,
+                        before_send=before_send)
+        with sentry_sdk.configure_scope() as scope:
+            exec_env = 'native'
+            # special variable set in the container
+            if os.getenv('IS_DOCKER_8395080871', 0):
+                # based on https://stackoverflow.com/a/42674935/616300
+                with open('/proc/1/cgroup', 'rt') as ifh:
```

Comment:
Maybe more succinct using `pathlib.Path` (this would replace lines 286-290)?

```suggestion
                exec_env = 'docker' if 'docker' in Path('/proc/1/cgroup').read_text() \
                    else 'singularity'
```